src/server/gc_worker/compaction_filter.rs

```diff
@@ -193,17 +199,47 @@ impl CompactionFilterInitializer<RocksEngine> for RocksEngine {
             region_info_provider,
             #[cfg(any(test, feature = "failpoints"))]
             callbacks_on_drop: vec![],
+            tablet_factory,
         });
     }
 }
 
-pub struct WriteCompactionFilterFactory;
+pub trait RocksdbCompactionFilterFactory: CompactionFilterFactory {
+    fn get_rocksdb(&self, region_id: u64, suffix: u64) -> engine_traits::Result<RocksEngine> {
```

Comment:
Just move these code into `create_compaction_filter`. A trait seems unnecessary.

Reply:
The only purpose is to eliminate dup code for create_compaction_filter for both raw and write compaction filter factories.

Reply:
Then a function is enough, because you didn't use `self` here.

Reply:
Good catch~ Rust compiler doesn't warm me.